src/ChainSafe.Gaming.Unity.MetaMask/MetaMaskProvider.cs

```diff
@@ -88,8 +88,9 @@ public override async Task<string> Connect()
                 EventName = "Metamask WebGL Initialized",
                 PackageName = "io.chainsafe.web3-unity",
             });
-
-            return await metaMaskController.Connect(chainConfig, chainRegistryProvider);
+            var connect = await metaMaskController.Connect(chainConfig, chainRegistryProvider);
+            await AddNetworkIfNotExistInWallet();
```

Comment:
this was already implemented in EthereumWindow.jslib, it wasn't working?

Reply:
You are right. I don't know how it worked tho, I will leave this only just in Reown then.